packages/dropcaps/init.lua

```diff
@@ -9,6 +9,18 @@ function package:_init ()
   self:loadPackage("raiselower")
 end
 
+local function getToleranceDepth ()
+  -- In non-strict mode, we allow using more lines to fit the dropcap.
+  -- However we cannot just check if the extra depth of the dropcap is above 0.
+  --  - our depth adjustment is but a best attempt.
+  --  - Some characters may have a small depth of their own, such as the "O" in Gentium Plus.
+  -- We must just ensure they stay within reasonable bounds with respect to the baseline,
+  -- so as not to flow over the next lines.
+  -- With well-formed text font, a good compromise is around 0.3bs (LaTeX does this too for defining its \strut).
+  -- This could be computed more precisely (e.g. computing the ratio between ascender and descender font metrics).
+  return SILE.measurement("0.3bs"):tonumber()
```

Comment:
What do you think about making the *relative* measurement a setting (so a user could tweak it to play nice with a weird font without struggling to reach this private function and only using this function (or probably just its calling location) to absolutize the measurement returned by the setting?

Reply:
I dunno, actually.
Another way would be to use the font metrics to compute the real value, and we actually have the "fontmetrics" C code based on harfbuzz and currently used by the linespascing package -- but I don't understand why we have it (i.e. what is its status besides our "justenoughharbuzz"?).
I did a quick test, assuming EB Garamond for the main text, it gives a 0.36(..) bs ratio for it, so "quite "close"
Perhaps we should go for the real things here? I.e. compute the real ascender / descender ratio (assuming font designers set those rights in their metrics) to avoid the hardcoded 0.3bs? We'd still need to round it a bit down, though, for safety (it it's _exactly_ the ratio, then we'll still have lines pretty closed, unless document.lineskip ticks in, but this is fragile)

Reply:
The "why we have it" for any of the `justenough*` libraries is simply glue. Those are all just thin wrappers around external binary libraries giving access to the functions we want to reach from Lua. That's it. Really they should all go away in favor of using full binding libraries, but since full binding sets don't always exist in Lua Simon built out "just enough" of a binding set to do the work we wanted to do.

For now we'll keep maintaining however many bindings we actually need in SILE, no more no less. It isn't a lot of work to keep around or even add new ones as needed.

So yes if you want to use font metrics for this we can do so, and if we need more functions we can add them.

This will be even easier down the road from the Rust side of things where there are more full libraries available with bindings. For example we won't need just enough HarfBuzz because we'll have 100% access to the HB API out of the box.

While using the volues Font designers encoded is the *right* thing to do, there will be lots and lots of cases where the values are invalid. Special decorative one-off projects are even more likely to be broken than the rest. Having some way to set this manually even if we computer a theoretically correct value might still be worth the effort.